packages/react-utilities/src/utils/applyTriggerPropsToChildren.ts

```diff
@@ -11,8 +11,25 @@ export const applyTriggerPropsToChildren = <TTriggerProps>(
   if (typeof children === 'function') {
     return children(triggerProps);
   } else if (children) {
-    return React.cloneElement(onlyChild(children), triggerProps);
+    return cloneTriggerTree(children, triggerProps);
   }
 
   return children;
 };
+
+/**
+ * Clones a React element tree, and applies the given props to the first grandchild that is not
+ * a FluentTriggerComponent or React Fragment (the same element returned by {@link getTriggerChild}).
+ */
+const cloneTriggerTree = <TTriggerProps>(child: React.ReactNode, triggerProps: TTriggerProps): React.ReactElement => {
+  if (!React.isValidElement(child)) {
+    throw new Error(`Component's child must be a single element`);
```

Comment:
Can you please improve this message a bit to be more helpful for customers?

```suggestion
    throw new Error(`A trigger element must be a single element for this component. Please ensure that you're not using React Fragments.`);
```

Something like that, for example.

---

Is there a reason to not use `React.Children.only()`? It will handle this case.

Reply:
It handles the `isValidElement` case, but not Fragments (which are valid elements). Since I need to have a check and custom error message for Fragments, it seems better to just have a single error message for both cases.